src/bosh_azure_cpi/lib/cloud/azure/models/vm_cloud_props.rb

```diff
@@ -28,26 +32,8 @@ def initialize(vm_properties, global_azure_config)
       @vm_properties = vm_properties.dup
 
       @instance_type = vm_properties['instance_type']
-      @resource_group_name = vm_properties.fetch('resource_group_name', global_azure_config.resource_group_name)
-      @storage_account_type = vm_properties['storage_account_type']
-      @availability_zone = vm_properties['availability_zone']
-      @availability_set = _parse_availability_set_config(vm_properties, global_azure_config)
-      @storage_account_name = vm_properties['storage_account_name']
-      @storage_account_max_disk_number = vm_properties.fetch('storage_account_max_disk_number', 30)
-      @storage_account_kind = vm_properties.fetch('storage_account_kind', STORAGE_ACCOUNT_KIND_GENERAL_PURPOSE_V1)
-      @tags = vm_properties.fetch('tags', {})
-      @caching = vm_properties.fetch('caching', 'ReadWrite')
-
-      @ip_forwarding = vm_properties['ip_forwarding']
-      @accelerated_networking = vm_properties['accelerated_networking']
-      @assign_dynamic_public_ip = vm_properties['assign_dynamic_public_ip']
-
-      @load_balancer = _parse_load_balancer_config(vm_properties, global_azure_config)
-
-      @application_gateway = vm_properties['application_gateway']
-
-      @application_security_groups = vm_properties['application_security_groups']
-      @security_group = Bosh::AzureCloud::SecurityGroup.parse_security_group(vm_properties['security_group'])
+      @instance_types = vm_properties['instance_types']
+      cloud_error("You need to specify one of 'vm_type/instance_type' or 'vm_resources'.") if @instance_type.nil? && (@instance_types.nil? || @instance_types.empty?)
```

Comment:
the "@instance_type" can be empty?

Reply:
yes. If `@instance_type` is empty, then users need to use `vm_resources`. And CPI will generate `@instance_types`.